WalletWasabi.Gui/Global.cs

```diff
@@ -518,24 +523,13 @@ public async Task InitializeWalletServiceAsync(KeyManager keyManager)
 					await Task.Delay(100, token);
 				}
 
-				WalletService = new WalletService(BitcoinStore, keyManager, Synchronizer, Nodes, DataDir, Config.ServiceConfiguration, FeeProviders, BitcoinCoreNode);
-
-				Logger.LogInfo($"Starting {nameof(WalletService)}...");
-				await WalletService.StartAsync(token);
-				Logger.LogInfo($"{nameof(WalletService)} started.");
-
-				token.ThrowIfCancellationRequested();
-
-				TransactionBroadcaster.AddWalletService(WalletService);
-				CoinJoinProcessor.AddWalletService(WalletService);
-
-				WalletService.TransactionProcessor.WalletRelevantTransactionProcessed += TransactionProcessor_WalletRelevantTransactionProcessed;
-				WalletService.ChaumianClient.OnDequeue += ChaumianClient_OnDequeued;
+				var walletService = new WalletService(BitcoinStore, keyManager, Synchronizer, Nodes, DataDir, Config.ServiceConfiguration, FeeProviders, BitcoinCoreNode);
```

Comment:
IMO here the api can be simpler. The wallet manager itself can new up the WalletService for you, and you simply do... `var walletService = await WalletManager.CreateAndStartAsync (krymanager ... etc)`

Or do you sometimes not want to start the Wallet immediately?

Just means you dont have to have both steps repeatedly, in UI Viewmodels, RPC code, Daemon, just a single api call.

Reply:
Furthermore, this can be improved in a way that to pass singleton instances in the WalletManager constructor so when you are adding wallet you have to pass only the key manager. Because of the size of the PR, I would postpone the conversation about this too into another PR.